engine/src/main/java/io/zeebe/engine/processor/workflow/deployment/model/transformer/CatchEventTransformer.java

```diff
@@ -74,22 +80,51 @@ private void transformMessageEventDefinition(
   }
 
   private void transformTimerEventDefinition(
+      final ExpressionLanguage expressionLanguage,
       final ExecutableCatchEventElement executableElement,
       final TimerEventDefinition timerEventDefinition) {
-    final Timer timer;
 
+    final Expression expression;
+    final TimerType type;
     if (timerEventDefinition.getTimeDuration() != null) {
       final String duration = timerEventDefinition.getTimeDuration().getTextContent();
-      timer = new RepeatingInterval(1, Interval.parse(duration));
+      expression = expressionLanguage.parseExpression(duration);
+      type = TimerType.DURATION;
     } else if (timerEventDefinition.getTimeCycle() != null) {
       final String cycle = timerEventDefinition.getTimeCycle().getTextContent();
-      timer = RepeatingInterval.parse(cycle);
+      expression = expressionLanguage.parseExpression(cycle);
+      type = TimerType.CYCLE;
     } else {
       final String timeDate = timerEventDefinition.getTimeDate().getTextContent();
-      timer = TimeDateTimer.parse(timeDate);
+      expression = expressionLanguage.parseExpression(timeDate);
+      type = TimerType.TIME_DATE;
     }
 
-    executableElement.setTimer(timer);
+    executableElement.setTimerFactory(
+        (expressionProcessor, scopeKey) -> {
+          switch (type) {
+            case DURATION:
+              return Optional.of(
+                      expressionProcessor.evaluateIntervalExpression(expression, scopeKey))
+                  .map(right -> new RepeatingInterval(1, right))
+                  .orElseThrow();
+            case CYCLE:
+              return Optional.of(expressionProcessor.evaluateStringExpression(expression, scopeKey))
```

Comment:
We should also support FEEL expressions using a `cycle` function. For example: `cycle(3, duration("PT1H"))`